cdc/puller/sorter/heap_sorter.go

```diff
@@ -155,7 +157,7 @@ func (h *heapSorter) flush(ctx context.Context, maxResolvedTs uint64) error {
 			backEnd := task.GetBackEnd()
 			if backEnd != nil {
 				defer task.markDeallocated()
-				return pool.dealloc(backEnd)
+				return h.pool.dealloc(backEnd)
```

Comment:
The panic fixed in this PR is `nil pointer dereference`, it seems caused by accessing global `pool` after it is destroyed (by calling `UnifiedSorterCleanUp()`), is there log `Unified Sorter: starting cleaning up files` before panic? Besides after this PR, the `h.pool` may still be `nil` here.

Reply:
The panic is found in std_err.log so I'm not sure if there are any `Unified Sorter: starting cleaning up files` before the panic. 

> Besides after this PR, the h.pool may still be nil here.

Could you elaborate? `h.pool` should is guaranteed to not be nil when it is assigned.

Reply:
What I mean is `h.pool` could be set to `nil` in `UnifiedSorterCleanUp()`

Maybe it is same with the old logic, the old code works as following

- Create unified sorter by `NewUnifiedSorter`
- Then run unified sorter, in the `Run` of unified sorter, heap sorter is initialized and this is where `flush` can be called. At this point, the global pool has been initialized.